client/src/partials/patient_edit/edit_patient.js

```diff
@@ -189,9 +189,23 @@ angular.module('bhima.controllers')
       customValidation();
     });
 
+    // Restart the search
+    $scope.restartSearch = function () {
+      originalPatientData = null;
+      $scope.patient = { origin_location_id: null, current_location_id: null };
+      $scope.initialOriginLocation = null;
+      $scope.initialCurrentLocation = null;
+
+      // Make sure the patient query gets processed
+      if ('processed' in dependencies.patient) {
```

Comment:
There is actually a feature for this.  You can call `validate.refresh(dependencies, ['patient'])` to "refresh" a dependency.  It works for the initial load as well, so if you use the same method for loading and reloading, you can use refresh.